recipe/meta.yaml

```diff
@@ -13,131 +12,135 @@ source:
+    - cling-patches/0052-Make-sure-no-EmittedDeferredDecls-are-left.patch
+    - cling-patches/0053-Keep-DeferredDecl-entry-when-forgetting-symbols-from.patch
+    - cling-patches/0054-Move-addEmittedDeferredDecl-into-separate-function-a.patch
+    - cling-patches/0055-Pass-mangled-name-to-CodeGenerator-forgetDecl.-From-.patch
+    - cling-patches/0056-Clear-also-EmittedDeferredDecls.patch
+    - cling-patches/0057-Also-remember-ODROnce-vars-fixes-cling-test-Prompt-S.patch
+    - cling-patches/0058-Keep-into-account-if-files-were-virtual.patch
+    - cling-patches/0059-Get-the-file-name-for-the-symbol-from-the-Module-not.patch
+    - cling-patches/0060-backport-r316022-Expose-ConsumeAnyToken-interface-to.patch
+    - cling-patches/0061-Backport-D45921.patch
+    - cling-patches/0062-D51295-Allow-resetting-of-NumCreatedFIDsForFileID.patch
 {% endif %}
 
 build:
   number: {{ build_number }}
-  string: {{ clang_variant }}_{{ build_number  }}
-  skip: true  # [py<36]
-
-{% if clang_variant == "cling" %}
-  features:
-    - cling
-    - vc14    # [win and py36]
-{% else %}
-  features:   # [win and py36]
-    - vc14    # [win and py36]
+  skip: true  # [win and vc<14]
+
+outputs:
+
+  # Main output
+  - name: clangdev
+    version: {{ version }}
+    script: build_clangdev.sh   # [unix]
+    script: build_clangdev.bat  # [win]
+    build:
+      activate_in_script: True
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - cmake >=3.4.3
+        # Needed to unpack the source tarball
+        - m2w64-xz  # [win]
+        # "compiling .pyc files" fails without this
+        - python >3
+      host:
+        - llvmdev =={{ version }}
+        - libxml2
+      run:
+        - llvmdev =={{ version }}
+        - libxml2
+        # Require the mutex package.
+        - clang_variant * {{ clang_variant }}
+    test:
+      requires:
+        - {{ compiler('cxx') }}
+        - cmake >=3.4.3
+      files:
+        - mytest.c
+      commands:
+        - $PREFIX/bin/clang --version  # [unix]
+        - $PREFIX/bin/clang++ --version  # [unix]
+        - $PREFIX/bin/clang-check --version  # [unix]
+
+        - $PREFIX/bin/clang -v -c mytest.c      # [unix]
+        - "%LIBRARY_BIN%/clang -v -c mytest.c"  # [win]
+
+        - if not exist %LIBRARY_INC%\\clang\\Lex\\Lexer.h exit 1  # [win]
+        - test -f $PREFIX/include/clang/Lex/Lexer.h      # [unix]
+        - test -f $PREFIX/lib/libclangBasic.a            # [unix]
+
+  # Mutex package to keep only one variant of clangdev in a given environment.
+  #
+  # Non-default variants track the "non_default_clang" feature.
+  # This is used to weigh down non-default variants of the package, allowing
+  # the default variant to take precedence unless required explicitely.
+  - name: clang_variant
+    version: 1.0
+    build:
+      string: {{ clang_variant }}
+{% if clang_variant != "default" %}
+      track_features:
+        - non_default_clang
```

Comment:
Would it be possible to not use `track_features` here?

Reply:
In fact there is no such feature, and no package has a `features` section.

 This was suggested by @msarahan on gitter. The `track_features` statement is just meant to weigh down the cling variant so that it is not installed by default.

This is used e.g. in the default channel for the selection of openblas vs mkl:
https://github.com/AnacondaRecipes/openblas-feedstock/blob/master/recipe/meta.yaml#L136

Reply:
Right, this "unmatched" `track_features` approach is now our best approach for specifying a variant preference. Because it doesn't have an accompanying `features` it avoids the messiness.

Reply:
How do you anticipate selecting the non-default variant? By pinning `clang_variant=*=cling`? My recommendation is to create a metapackage that depends on `clang_variant=*=cling` that users can install. That's a cleaner user experience than pinning, in my view.

Reply:
I was thinking of cling itself doing `clang_variant=*=cling` in its runtime dependencies. I can't think of a situation where we would want this variant and not cling.

Reply:
> Right, this "unmatched" track_features approach is now our best approach for specifying a variant preference. Because it doesn't have an accompanying features it avoids the messiness.

Since features are meant to disappear, it could be interesting to replace the unmatched `track_features` approach with some other functionally equivalent `low_precedence: true` thingy.

Reply:
...or better yet `precedence: <some number>` 😉